addons/product_expiry/models/stock_move_line.py

```diff
@@ -9,19 +9,18 @@
 class StockMoveLine(models.Model):
     _inherit = "stock.move.line"
 
-    expiration_date = fields.Datetime(string='Expiration Date',
+    expiration_date = fields.Datetime(string='Expiration Date', compute='_compute_expiration_date', store=True,
         help='This is the date on which the goods with this Serial Number may'
         ' become dangerous and must not be consumed.')
-
-    @api.onchange('product_id', 'product_uom_id')
-    def _onchange_product_id(self):
-        res = super(StockMoveLine, self)._onchange_product_id()
-        if self.picking_type_use_create_lots:
-            if self.product_id.use_expiration_date:
-                self.expiration_date = fields.Datetime.today() + datetime.timedelta(days=self.product_id.expiration_time)
-            else:
-                self.expiration_date = False
-        return res
+    
+    @api.depends('product_id')
+    def _compute_expiration_date(self):
+        for record in self:
```

Comment:
```suggestion
        for move_line in self:
```
Comme ça on sait sur quoi on travaille c'est plus facile a lire je trouves